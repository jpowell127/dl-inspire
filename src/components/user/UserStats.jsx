import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { Statistic } from "semantic-ui-react";

function UserStats({ stats }) {
  const { followers, following, appreciations, views, comments } = stats;

  return (
    <Statistic.Group size="mini" widths="five">
      <Statistic color="red">
        <Statistic.Value>{followers}</Statistic.Value>
        <Statistic.Label size="mini">Followers</Statistic.Label>
      </Statistic>
      <Statistic color="orange">
        <Statistic.Value>{following}</Statistic.Value>
        <Statistic.Label>Following</Statistic.Label>
      </Statistic>
      <Statistic color="yellow">
        <Statistic.Value>{appreciations}</Statistic.Value>
        <Statistic.Label>Appreciations</Statistic.Label>
      </Statistic>
      <Statistic color="green">
        <Statistic.Value>{views}</Statistic.Value>
        <Statistic.Label>Views</Statistic.Label>
      </Statistic>
      <Statistic color="teal">
        <Statistic.Value>{comments}</Statistic.Value>
        <Statistic.Label>Comments</Statistic.Label>
      </Statistic>
    </Statistic.Group>
  );
}

UserStats.propTypes = {
  stats: PropTypes.shape({
    followers: PropTypes.number.isRequired,
    following: PropTypes.number.isRequired,
    appreciations: PropTypes.number.isRequired,
    views: PropTypes.number.isRequired,
    comments: PropTypes.number.isRequired
  }).isRequired
};

export default UserStats;
