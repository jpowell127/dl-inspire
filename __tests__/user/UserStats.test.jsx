import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import UserStats from "../../src/components/user/UserStats";

test("UserStats renders correctly", () => {
  const stats = {
    followers: 1,
    following: 1,
    appreciations: 1,
    views: 1,
    comments: 1
  };

  const component = shallow(<UserStats stats={stats} />);
  expect(component).toMatchSnapshot();
});
