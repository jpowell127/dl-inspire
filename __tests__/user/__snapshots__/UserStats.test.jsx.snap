// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserStats renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserStats
    stats={
      Object {
        "appreciations": 1,
        "comments": 1,
        "followers": 1,
        "following": 1,
        "views": 1,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Statistic
          color="red"
        >
          <StatisticValue>
            1
          </StatisticValue>
          <StatisticLabel
            size="mini"
          >
            Followers
          </StatisticLabel>
        </Statistic>,
        <Statistic
          color="orange"
        >
          <StatisticValue>
            1
          </StatisticValue>
          <StatisticLabel>
            Following
          </StatisticLabel>
        </Statistic>,
        <Statistic
          color="yellow"
        >
          <StatisticValue>
            1
          </StatisticValue>
          <StatisticLabel>
            Appreciations
          </StatisticLabel>
        </Statistic>,
        <Statistic
          color="green"
        >
          <StatisticValue>
            1
          </StatisticValue>
          <StatisticLabel>
            Views
          </StatisticLabel>
        </Statistic>,
        <Statistic
          color="teal"
        >
          <StatisticValue>
            1
          </StatisticValue>
          <StatisticLabel>
            Comments
          </StatisticLabel>
        </Statistic>,
      ],
      "size": "mini",
      "widths": "five",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <StatisticValue>
              1
            </StatisticValue>,
            <StatisticLabel
              size="mini"
            >
              Followers
            </StatisticLabel>,
          ],
          "color": "red",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Followers",
              "size": "mini",
            },
            "ref": null,
            "rendered": "Followers",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <StatisticValue>
              1
            </StatisticValue>,
            <StatisticLabel>
              Following
            </StatisticLabel>,
          ],
          "color": "orange",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Following",
            },
            "ref": null,
            "rendered": "Following",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <StatisticValue>
              1
            </StatisticValue>,
            <StatisticLabel>
              Appreciations
            </StatisticLabel>,
          ],
          "color": "yellow",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Appreciations",
            },
            "ref": null,
            "rendered": "Appreciations",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <StatisticValue>
              1
            </StatisticValue>,
            <StatisticLabel>
              Views
            </StatisticLabel>,
          ],
          "color": "green",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Views",
            },
            "ref": null,
            "rendered": "Views",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <StatisticValue>
              1
            </StatisticValue>,
            <StatisticLabel>
              Comments
            </StatisticLabel>,
          ],
          "color": "teal",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": 1,
            },
            "ref": null,
            "rendered": 1,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Comments",
            },
            "ref": null,
            "rendered": "Comments",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Statistic
            color="red"
          >
            <StatisticValue>
              1
            </StatisticValue>
            <StatisticLabel
              size="mini"
            >
              Followers
            </StatisticLabel>
          </Statistic>,
          <Statistic
            color="orange"
          >
            <StatisticValue>
              1
            </StatisticValue>
            <StatisticLabel>
              Following
            </StatisticLabel>
          </Statistic>,
          <Statistic
            color="yellow"
          >
            <StatisticValue>
              1
            </StatisticValue>
            <StatisticLabel>
              Appreciations
            </StatisticLabel>
          </Statistic>,
          <Statistic
            color="green"
          >
            <StatisticValue>
              1
            </StatisticValue>
            <StatisticLabel>
              Views
            </StatisticLabel>
          </Statistic>,
          <Statistic
            color="teal"
          >
            <StatisticValue>
              1
            </StatisticValue>
            <StatisticLabel>
              Comments
            </StatisticLabel>
          </Statistic>,
        ],
        "size": "mini",
        "widths": "five",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <StatisticValue>
                1
              </StatisticValue>,
              <StatisticLabel
                size="mini"
              >
                Followers
              </StatisticLabel>,
            ],
            "color": "red",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Followers",
                "size": "mini",
              },
              "ref": null,
              "rendered": "Followers",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <StatisticValue>
                1
              </StatisticValue>,
              <StatisticLabel>
                Following
              </StatisticLabel>,
            ],
            "color": "orange",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Following",
              },
              "ref": null,
              "rendered": "Following",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <StatisticValue>
                1
              </StatisticValue>,
              <StatisticLabel>
                Appreciations
              </StatisticLabel>,
            ],
            "color": "yellow",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Appreciations",
              },
              "ref": null,
              "rendered": "Appreciations",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <StatisticValue>
                1
              </StatisticValue>,
              <StatisticLabel>
                Views
              </StatisticLabel>,
            ],
            "color": "green",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Views",
              },
              "ref": null,
              "rendered": "Views",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <StatisticValue>
                1
              </StatisticValue>,
              <StatisticLabel>
                Comments
              </StatisticLabel>,
            ],
            "color": "teal",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Comments",
              },
              "ref": null,
              "rendered": "Comments",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
